app/src/main/java/swati4star/createpdf/util/PdfToImages.java

```diff
@@ -26,35 +28,61 @@
     private ExtractImagesListener mExtractImagesListener;
     private int mImagesCount = 0;
     private ArrayList<String> mOutputFilePaths;
+    private String[] mPassword;
+    private PDFEncryptionUtility mPDFEncryptionUtility;
+    private Context mContext;
+    private String mDecryptedPath;
 
-    public PdfToImages(String mPath, Uri mUri, ExtractImagesListener mExtractImagesListener) {
+    public PdfToImages(Context context, String mPath, Uri mUri,
+                       ExtractImagesListener mExtractImagesListener) {
         this.mPath = mPath;
         this.mUri = mUri;
         this.mExtractImagesListener = mExtractImagesListener;
         mOutputFilePaths = new ArrayList<>();
+        this.mContext = context;
+    }
+
+    public PdfToImages(Context context, String[] password, String mPath, Uri mUri,
+                       ExtractImagesListener mExtractImagesListener) {
+        this.mPath = mPath;
+        this.mUri = mUri;
+        this.mExtractImagesListener = mExtractImagesListener;
+        mOutputFilePaths = new ArrayList<>();
+        this.mPassword = password;
+        this.mContext = context;
     }
 
     @Override
     protected void onPreExecute() {
         super.onPreExecute();
+        mPDFEncryptionUtility = new PDFEncryptionUtility((Activity) mContext);
         mExtractImagesListener.extractionStarted();
     }
 
     @Override
     protected Void doInBackground(Void... voids) {
+        if (mPassword != null) {
```

Comment:
If you are checking here `mPassword != null` then there is really a need of another constructor and checking at line `229` of `PdfToImageFragment` ?